### PR TITLE
GCPClusterTemplate CRD added

### LINKED ISSUE #397: Add GCPClusterTemplate type
/kind feature

**Describe the solution you'd like:**
Add a new CRD named GCPClusterTemplate. This is needed for ClusterClass.
In the short term this type will not be used outside of ClusterClass.

### PR DESCRIPTION
**Pull request by shivi28**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR adds GCPClusterTemplate type needed for the ClusterClass proposal.
In the near future, this type would only be used by ClusterClass.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #397 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Add GCPClusterTemplate type
```

### CODE REVIEW

File: PROJECT [hunk 1]
```diff
@@ -20,3 +20,6 @@ resources:
 - group: infrastructure
   version: v1alpha4
   kind: GCPMachineTemplate
+- group: infrastructure
+  version: v1alpha4
+  kind: GCPClusterTemplate
```
Comment (cpanato) [inline PROJECT:25]: nit: new line

File: api/v1alpha4/gcpclustertemplate_webhook.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+var _ webhook.Defaulter = &GCPClusterTemplate{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type.
+func (r *GCPClusterTemplate) Default() {
+	gcpclustertemplatelog.Info("default", "name", r.Name)
+}
+
+//+kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha4-gcpclustertemplate,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=gcpclustertemplates,versions=v1alpha4,name=validation.gcpclustertemplate.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1beta1
+
+var _ webhook.Validator = &GCPClusterTemplate{}
```
Comment (cpanato) [inline api/v1alpha4/gcpclustertemplate_webhook.go:50]: is it possible to add some tests for this? similar we have for CAPZ? https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1501

Comment (shivi28) [reply api/v1alpha4/gcpclustertemplate_webhook.go:50]: yeah, added for validateUpdate. No validation was done during creation and deletion, so skipping them.

**Review by cpanato — APPROVED** [review]
thanks for this!

/lgtm
/approve

**Pull request merged**